components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/NettyConnection.java

```diff
@@ -52,26 +54,33 @@
     /**
      * Constructs an instance with an arbitrary UUID.
      *
-     * @param origin  the origin connected to
-     * @param channel the netty channel used
+     * @param origin                  the origin connected to
+     * @param channel                 the netty channel used
      * @param requestOperationFactory used to create operation objects that send http requests via this connection
+     * @param httpConfig              configuration settings for the <b>origin</b>
+     * @param sslContext              TLS context in case of secure connections
+     * @param sendSni                 include the servername extension (server name indicator) in the TLS handshake
+     * @param sniHost                 hostname override for the server name indicator
      */
-    public NettyConnection(Origin origin, Channel channel, HttpRequestOperationFactory requestOperationFactory, HttpConfig httpConfig, SslContext sslContext) {
+    public NettyConnection(Origin origin, Channel channel, HttpRequestOperationFactory requestOperationFactory,
+                           HttpConfig httpConfig, SslContext sslContext, boolean sendSni, Optional<String> sniHost) {
         this.origin = requireNonNull(origin);
         this.channel = requireNonNull(channel);
         this.requestOperationFactory = requestOperationFactory;
         this.channel.pipeline().addLast(new TimeToFirstByteHandler(this));
         this.channel.closeFuture().addListener(future ->
                 listeners.announce().connectionClosed(NettyConnection.this));
-
-        addChannelHandlers(channel, httpConfig, sslContext);
+        addChannelHandlers(channel, httpConfig, sslContext, sendSni, sniHost.orElse(origin.host()));
     }
 
-    private static void addChannelHandlers(Channel channel, HttpConfig httpConfig, SslContext sslContext) {
+    private static void addChannelHandlers(Channel channel, HttpConfig httpConfig, SslContext sslContext, boolean sendSni, String targetHost) {
         ChannelPipeline pipeline = channel.pipeline();
 
         if (sslContext != null) {
-            pipeline.addLast("ssl", sslContext.newHandler(channel.alloc()));
+            SslHandler sslHandler = sendSni
+                    ? sslContext.newHandler(channel.alloc(), targetHost, -1)
```

Comment:
Is it correct to pass -1 as a peer port? Should we instead pass in the configured port number?

Reply:
Also, is it necessary to test for `sendSni` flag given that the call site has already set the target accordingly? This is from the `NettyConnection` constructor: 

    sniHost.orElse(origin.host())

Reply:
The sendSni flag is to disable sni,  whereas the 'orElse' is to send the target hostname in SNI  instead of a preconfigured override.

It seems that -1 is a valid value as it is the one that will be set in the code with no SNI. The SNI  extension doesn't specify a port number but only a server name.  It's a shame that netty does not have an overload without a port number.